apps/dashboard/app/controllers/projects_controller.rb

```diff
@@ -110,9 +110,34 @@ def job_details
 
     hpc_job = project.job(job_details_params[:jobid].to_s, cluster_str)
 
+    @project = project
     render(partial: 'job_details', locals: { job: hpc_job })
   end
 
+  # DELETE /projects/:project_id/jobs/:cluster/:jobid
+  def delete_job
+    project = Project.find(job_details_params[:project_id])
+    cluster_str = job_details_params[:cluster].to_s
+    hpc_job = project.job(job_details_params[:jobid].to_s, cluster_str)
+
+    if hpc_job.status.to_s == 'completed'
+      redirect_to project_path(job_details_params[:project_id]) if hpc_job.status.to_s == 'completed'
+    else
+      project.remove_logged_job(job_details_params[:jobid].to_s, cluster_str)
+      redirect_to project_path(job_details_params[:project_id])
+    end
+  end
+
+  # PATCH /projects/:project_id/jobs/:cluster/:jobid/stop
+  def stop_job
+    cluster_str = job_details_params[:cluster].to_s
+    cluster = OodAppkit.clusters[cluster_str.to_sym]
+
+    cluster.job_adapter.delete(job_details_params[:jobid].to_s)
```

Comment:
Seems like this can fail. Like we should rescue a StandardError here just in case it does.

Reply:
It definitely can. Sure the UI may not allow for such a case, but anyone can enter anything into a browsers URL or use curl and encounter the error with a bad jobid.